MSVC编译器报错“文本后缀无效”

MSVC编译ch21/word_query.cpp报错：

```
MSBuild version 17.4.1+9a89d02ff for .NET Framework
  fltk.vcxproj -> D:\CLionProjects\PPP-code\cmake-build-debug-visual-studio-2022\_deps\fltk-build\lib\Debug\fltkd.lib
  word_query.cpp
D:\CLionProjects\PPP-code\ch21\word_query.cpp(1,1): warning C4819: 该文件包含不能在当前代码页(936)中表示的字符。请将该文件保存为 Unicode 格式以防止数据丢失 [D:\CLionProjects\PPP-code\cmake-build-debug-visual-studio-2022\ch21\word_query.vcxproj]
D:\CLionProjects\PPP-code\ch21\word_query.cpp(103,88): error C3688: 文本后缀“��|�”无效；未找到文文本运算符或文本运算符模板“operator """"��|�” [D:\CLionProjects\PPP-code\cmake-build-debug-visual-studio-2022\ch21\word_query.vcxproj]
D:\CLionProjects\PPP-code\ch21\word_query.cpp(105,37): error C2065: “RE_SINGLE_QUOTES”: 未声明的标识符 [D:\CLionProjects\PPP-code\cmake-build-debug-visual-studio-2022\ch21\word_query.vcxproj]
```

相关代码如下：

```cpp
static const std::regex RE_DOUBLE_QUOTES("“|”"), RE_SINGLE_QUOTES("‘|’");
```

但是gcc编译器正常，GitHub Actions的build-visual-studio也正常。

怀疑是字符编码的问题：源代码是UTF-8编码，而MSVC默认使用GBK编码，如果源代码中包含中文字符（全角引号）就会编译失败。将源文件编码改为GBK后错误就消除了。

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,10 @@ include(testing.cmake)
 include(gui.cmake)
 
 include_directories(${CMAKE_SOURCE_DIR})  # 头文件包含可相对于根目录
+if(MSVC)
+  add_compile_options(/utf-8)
+endif()
+
 add_subdirectory(GUI)  # 作者提供的GUI库
 file(GLOB CHAPTERS RELATIVE ${CMAKE_SOURCE_DIR} ch*)
 foreach(c IN ITEMS ${CHAPTERS})

